src/raven.js

```diff
@@ -545,9 +549,12 @@ Raven.prototype = {
     _wrapBuiltIns: function() {
         var self = this;
 
-        function fill(obj, name, replacement) {
+        function fill(obj, name, replacement, noUndo) {
             var orig = obj[name];
             obj[name] = replacement(orig);
+            if (!noUndo) { // xhr instances can't undo otherwise mem leaks
```

Comment:
Let's just comment a bit more on _why_ this is a memory leak, and why it's unique to XHR.